packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -88,6 +88,37 @@ class CozyStackClient {
     return fetch(this.fullpath(path), options)
   }
 
+  /**
+   * Retrieves a new app token by refreshing the currently used token.
+   * @throws {Error} The client should already have an access token to use this function
+   * @throws {Error} The client couldn't fetch a new token
+   * @returns {Promise} A promise that resolves with a new AccessToken object
+   */
+  async refreshToken() {
+    if (!this.token) throw new Error('No token to refresh')
+
+    const options = {
+      method: 'GET',
+      credentials: 'include'
+    }
+    const response = await fetch('/', options)
+
+    if (response.ok) {
+      const html = await response.text()
+      const parser = new DOMParser()
+      const doc = parser.parseFromString(html, 'text/html')
+      const appNode = doc.querySelector('div[role="application"]')
+      const cozyToken = appNode.dataset.cozyToken
```

Comment:
Do we need some defensive code here, or are we absolutely sure that `/` will always have a `div[role="application"]` and a token ?

Reply:
Good catch, let's add some conditions.

Reply:
Yes, I think a check and an early return if appNode or dataset.cozyToken does not exist should be there.